test/e2e/utils_test.go

```diff
@@ -66,39 +79,58 @@ func verifyOperatorStatusCondition(client *certmanoperatorclient.Clientset, cont
 	return errors.NewAggregate(errs)
 }
 
-// removeOverrides removes all the overrides from all the cert-manager operands. The update process is retried if
-// a conflict error is encountered.
-func removeOverrides(client *certmanoperatorclient.Clientset) error {
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+// resetCertManagerState is used to revert back to the default cert-manager operands' state
+func resetCertManagerState(ctx context.Context, client *certmanoperatorclient.Clientset, loader library.DynamicResourceLoader) error {
+	// update operator spec to empty *Config and set operatorSpec to default values
+	err := wait.PollImmediate(PollInterval, TestTimeout, func() (done bool, err error) {
 		operator, err := client.OperatorV1alpha1().CertManagers().Get(context.TODO(), "cluster", v1.GetOptions{})
 		if err != nil {
-			return err
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
 		}
 
 		updatedOperator := operator.DeepCopy()
-
-		hasOverride := false
-		if updatedOperator.Spec.ControllerConfig != nil {
-			updatedOperator.Spec.ControllerConfig = nil
-			hasOverride = true
-		}
-		if updatedOperator.Spec.WebhookConfig != nil {
-			updatedOperator.Spec.WebhookConfig = nil
-			hasOverride = true
-		}
-		if updatedOperator.Spec.CAInjectorConfig != nil {
-			updatedOperator.Spec.CAInjectorConfig = nil
-			hasOverride = true
+		updatedOperator.Spec.CAInjectorConfig = nil
+		updatedOperator.Spec.ControllerConfig = nil
+		updatedOperator.Spec.WebhookConfig = nil
+		updatedOperator.Spec.OperatorSpec = opv1.OperatorSpec{
+			ManagementState: opv1.Managed,
 		}
 
-		if !hasOverride {
-			return nil
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err = client.OperatorV1alpha1().CertManagers().Update(context.TODO(), updatedOperator, v1.UpdateOptions{})
+			return err
+		})
```

Comment:
I think the get call needs to be inside `retry.RetryOnConflict `, or else the conflict may not get resolved.

Reply:
Cool, lemme update it then.

Reply:
I'm also verifying locally if I need to check the `*Config`'s first before setting them to `nil` i.e. if kube-apiserver returns an error if update has 0 changes in it. Will modify accordingly if required as well.

Reply:
> if update has 0 changes in it

just to confirm, this isn't a problem, apiserver can handle such updates without returning errors.

Reply:
> I think the get call needs to be inside retry.RetryOnConflict , or else the conflict may not get resolved.

@arkadeepsen Updated it now!